quap/document_stores/data_corpus_store.py

```diff
@@ -84,6 +86,33 @@ def remove_document(self, document: Document) -> None:
 
         # deletes all the contexts of the specified document
         self.delete_all_documents(index=document.corpus.contexts_index, filters={
-            '$and': {'document_name': {'$eq': document.name}}
+            '$and': {'name': {'$eq': document.name}}
         })
         self.delete_documents(ids=[str(document.id)], index=document.corpus.original_documents_index)
+
+    def add_dataset(self,
+                    preprocessed_docs: list[haystack.schema.Document],
+                    preprocessed_labels: list[haystack.Label],
+                    original_docs: list[haystack.schema.Document],
+                    preprocessed_docs_index: str,
+                    preprocessed_labels_index: str,
+                    original_docs_index: str
+                    ) -> None:
+        """
+        Add already preprocessed dataset documents and labels to DataCorpusStore.
+
+        :param preprocessed_docs: A list of documents generated by `haystack.document_stores.utils.eval_data_from_json`.
+        :param preprocessed_labels: A list of full documents corresponding to `preprocessed_docs`.
+        :param original_docs: A list of labels generated by `haystack.document_stores.utils.eval_data_from_json`.
+        :param preprocessed_docs_index: Str index (from DataCorpus).
+        :param preprocessed_labels_index: Str index (from Dataset).
+        :param original_docs_index: Str index (from DataCorpus).
+        :return:
+        """
+        self.delete_documents(index=original_docs_index)
```

Comment:
Tu trzeba dodać najpierw sprawdzenie czy ten indeks w ogóle istnieje

Reply:
Noo okej, ale to się nie psuje jak nie ma nic na indeksie.

Reply:
Problem może być nie z tym, że indeks pusty, a z tym, że indeks nie istnieje